### PR TITLE
Implement get_command_argument

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: #4593

### CODE REVIEW

**Comment by certik** [comment]
Looks great. 3rd party tests must be fixed.

**Comment by HarshitaKalani** [comment]
```
lfortran   -o gsnap global.o snap_main.o utils.o version.o plib.o geom.o sn.o data.o control.o input.o setup.o dealloc.o translv.o solvar.o outer.o expxs.o inner.o sweep.o octsweep.o dim1_sweep.o dim3_sweep.o output.o time.o mms.o analyze.o thrd_comm.o mkba_sweep.o
+ ./gsnap ../qasnap/sample/inp out
Runtime error: File `./gsnap ../qasnap/sample/inp out` does not exists!
Cannot open a file with the `status=old`
```

**Comment by gxyd** [comment]
It seems like a genuine error, e.g; when I execute the below Fortran program (with your implementation of `get_command_argument`:
```f90
program command_argument_example
    implicit none
    integer :: arg_num
    character(len=100) :: arg_value

    ! Print a message if no arguments are passed
    call get_command_argument(0, arg_value)
    if (trim(arg_value) == "") then
        print *, "No command-line arguments provided."
    else
        ! Loop through the arguments passed on the command line
        do arg_num = 1, 3  ! Assuming we want to read 3 arguments
            call get_command_argument(arg_num, arg_value)
            if (trim(arg_value) /= "") then
                print *, "Argument ", arg_num, ": ", trim(arg_value)
            else
                print *, "No more arguments."
                exit
            end if
        end do
    end if
end program command_argument_example

```

And I run it as:
```
$ lfortran -o test test.f90
$ ./test arg1 arg2 arg3                                 lf 04:29:35 PM
Argument     1    :     ./test arg1 arg2 arg3
Argument     2    :     ./test arg1 arg2 arg3
Argument     3    :     ./test arg1 arg2 arg3
```

I get output as all the arguments (along with the "object file" as well), while the expected output is:
```
 Argument            1 : arg1
 Argument            2 : arg2
 Argument            3 : arg3
```

i.e. invidually listing all the arguments (only one).

It would be pretty good if you can also add some test cases to `get_command_argument` function.

**Comment by gxyd** [comment]
I've moved the PR to draft, as it needs some work currently for it to be mergable.

**Comment by gxyd** [comment]
> ```
> lfortran   -o gsnap global.o snap_main.o utils.o version.o plib.o geom.o sn.o data.o control.o input.o setup.o dealloc.o translv.o solvar.o outer.o expxs.o inner.o sweep.o octsweep.o dim1_sweep.o dim3_sweep.o output.o time.o mms.o analyze.o thrd_comm.o mkba_sweep.o
> + ./gsnap ../qasnap/sample/inp out
> Runtime error: File `./gsnap ../qasnap/sample/inp out` does not exists!
> Cannot open a file with the `status=old`
> ```

Just like the above example, it's taking the filename (i.e. argument) as the whole command itself (i.e. including the object file being executed as well)

**Comment by HarshitaKalani** [comment]
Now it gives the correct output
```console
$ lfortran -o test integration_tests/intrinsics_317.f90 
$ ./test arg1 arg2 arg3
Argument     1    :     arg1
Argument     2    :     arg2
Argument     3    :     arg3
```

**Comment by HarshitaKalani** [comment]
This is now ready for a review

File: src/libasr/pass/intrinsic_subroutines.h [hunk 1]
```diff
@@ -285,6 +286,93 @@ namespace GetCommand {
...
+        if(x.n_args > 1) {
+            ASRUtils::require_impl(ASRUtils::is_character(*ASRUtils::expr_type(x.m_args[1])), "Second argument must be of character type", x.base.base.loc, diagnostics);
+        }
+        if (x.n_args > 2) {
+            ASRUtils::require_impl(ASRUtils::is_integer(*ASRUtils::expr_type(x.m_args[2])), "Third argument must be of integer type", x.base.base.loc, diagnostics);
+        } 
+        if (x.n_args == 4) {
+            ASRUtils::require_impl(ASRUtils::is_integer(*ASRUtils::expr_type(x.m_args[3])), "Fourth argument must be of integer type", x.base.base.loc, diagnostics);
+        } else {
+            ASRUtils::require_impl(false, "Unexpected number of args, get_command takes 3 arguments, found " + std::to_string(x.n_args), x.base.base.loc, diagnostics);
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/intrinsic_subroutines.h:304]:
```suggestion
            ASRUtils::require_impl(false, "Unexpected number of args, get_command takes atmost 4 arguments, found " + std::to_string(x.n_args), x.base.base.loc, diagnostics);
```

File: src/libasr/pass/intrinsic_subroutines.h [hunk 2]
```diff
@@ -285,6 +286,93 @@ namespace GetCommand {
...
+        declare_basic_variables(new_name);
+        Vec<ASR::expr_t*> call_args; call_args.reserve(al, 0);
+        fill_func_arg_sub("number", arg_types[0], In);
+        Vec<ASR::expr_t*> args_1; args_1.reserve(al, 0);
+        SymbolTable *fn_symtab_1 = al.make_new<SymbolTable>(fn_symtab);
+        ASR::expr_t *arg = b.Variable(fn_symtab_1, "n", arg_types[0],
+            ASR::intentType::In, ASR::abiType::BindC, true);
+        args_1.push_back(al, arg);
+        SetChar dep_1; dep_1.reserve(al, 1);
+        Vec<ASR::stmt_t*> body_1; body_1.reserve(al, 1);
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/intrinsic_subroutines.h:335]: Nit: If the above lines is only used inside the `if (arg_types.size() > 1)`, how about we move it inside the if block?

File: integration_tests/intrinsics_317.f90 [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    character(len=100) :: arg_value
+
+    ! Print a message if no arguments are passed
+    call get_command_argument(0, arg_value)
+    if (trim(arg_value) == "") then
+        print *, "No command-line arguments provided."
+    else
+        ! Loop through the arguments passed on the command line
+        do arg_num = 1, 3  ! Assuming we want to read 3 arguments
+            call get_command_argument(arg_num, arg_value)
```
Comment (Thirumalai-Shaktivel) [inline integration_tests/intrinsics_317.f90:13]: Can we also check the length and status in the get_command_argument?

**Review by Thirumalai-Shaktivel — COMMENTED** [review]
Otherwise, LGTM

**Review by certik — APPROVED** [review]
I think this looks good, once @Thirumalai-Shaktivel's feedback is addressed, we can merge it.

File: integration_tests/intrinsics_317.f90 (hunk 3 above)
Comment (HarshitaKalani) [reply integration_tests/intrinsics_317.f90]:
```
$ lfortran -o test integration_tests/intrinsics_317.f90 
$ ./test arg1 arg12 arg13
Argument     1    :     arg1
Length    4
Status    0
Argument     2    :     arg12
Length    5
Status    0
Argument     3    :     arg13
Length    5
Status    0
$ gfortran integration_tests/intrinsics_317.f90 && ./a.out arg1 arg12 arg13
 Argument            1 : arg1
 Length           4
 Status           0
 Argument            2 : arg12
 Length           5
 Status           0
 Argument            3 : arg13
 Length           5
 Status           0
 ```

**Review by certik — APPROVED** [review]

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Comment by certik** [comment]
I am going to merge it. If it fails on Windows after we fix https://github.com/lfortran/lfortran/issues/4919, then we'll just submit a PR to fix Windows, it shouldn't be difficult.

**Pull request merged**